db/system_distributed_keyspace.cc

```diff
@@ -239,7 +239,7 @@ static list_type_impl::native_type prepare_cdc_generation_description(const cdc:
         }
 
         ret.push_back(make_tuple_value(cdc_token_range_description_type,
-                { data_value(e.token_range_end.to_sstring())
+                { data_value(dht::token::to_le_int(e.token_range_end))
```

Comment:
Doesn't `data_value` take care of endianness already? It would be weird for it not to do that, because it's supposed to represent values in table cells.

A lot of tests suggest that this is the case, because they do a lot of stuff like executing
`update ... set v = 12345678 where ...`
followed by querying the table and comparing the value to
`data_value(int32_t(12345678))`
in C++.

Reply:
But I guess it doesn't matter in this case: even if the conversion is redundant, you undo it on deserialization.

Reply:
Oh, but it probably would matter if you deserialize on a machine which uses different endianness than the one which you serialize on.

Reply:
@kbr- why would a type of the machine matter? `to_le_int` converts cpu type to little endian, so if it's transferred over the network to the other machine, the machine would simply convert from little endian back to its native type. It's no different than any other integer serialization.

Reply:
Ok, it shouldn't matter.

Suppose that `data_value` takes care to store values in network order (big-endian). Then:
1. on BE machine: `to_le_int`/`from_le_int` = reverse, `data_value` cons/uncons = identity; composition = reverse
2. on LE machine: `to_le_int`/`from_le_int` = identity, `data_value` cons/uncons = reverse; composition = reverse

So with this patch, we would always reverse on serialization and deserialization.